### PR TITLE
Remove usage of robotInit from FRC Docs

### PR DESCRIPTION
**Pull request by DJB755**
Remove usage of robotInit #2679

### CODE REVIEW

**Comment by DJB755** [comment]
Oop realized I did not go throught the examples linked in the docs, will do that before submitting the PR

**Comment by DJB755** [comment]
Remove usage of robotInit https://github.com/wpilibsuite/frc-docs/issues/2679 replaced with different constructor on the FRCDocs side, should I fork allwpilib too and change the examples there too before?

File: source/docs/software/advanced-controls/trajectories/trajectory-generation.rst [hunk 1]
```diff
@@ -67,7 +67,7 @@ Here is an example of generating a trajectory using clamped cubic splines for th
 
 .. note:: The Java code utilizes the `Units <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/util/Units.html>`_ utility, for easy unit conversions.
 
-.. note:: Generating a typical trajectory takes about 10 ms to 25 ms. This isn't long, but it's still highly recommended to generate all trajectories on startup (``robotInit``).
+.. note:: Generating a typical trajectory takes about 10 ms to 25 ms. This isn't long, but it's still highly recommended to generate all trajectories on startup (``Robot``).
```
Comment (sciencewhiz) [inline source/docs/software/advanced-controls/trajectories/trajectory-generation.rst:70]: This should say something like ``Robot`` constructor

File: source/docs/software/basic-programming/alliancecolor.rst [hunk 2]
```diff
@@ -3,7 +3,7 @@ Get Alliance Color
...
+Note that there are three cases: red, blue, and no color yet.  It is important that code handles the third case correctly because the alliance color will not be available until the Driver Station connects.  In particular, code should not assume that the alliance color will be available during constructor methods or `Robot`, but it should be available by the time `autoInit` or `teleopInit` is called.  FMS will set the alliance color automatically; when not connected to FMS, the alliance color can be set from the Driver Station (see :ref:`"Team Station" on the Operation Tab <docs/software/driverstation/driver-station:Operation Tab>`).
```
Comment (sciencewhiz) [inline source/docs/software/basic-programming/alliancecolor.rst:6]: This shouldn't be changed, as now it's redundant

File: source/docs/software/basic-programming/reading-stacktraces.rst [hunk 3]
```diff
@@ -42,8 +42,8 @@ To start, search above the ``unexpected error has occurred`` for the stack trace
 
       .. code-block:: text
 
-         Error at frc.robot.Robot.robotInit(Robot.java:24): Unhandled exception: java.lang.NullPointerException
-                  at frc.robot.Robot.robotInit(Robot.java:24)
+         Error at frc.robot.Robot.Robot(Robot.java:24): Unhandled exception: java.lang.NullPointerException
```
Comment (sciencewhiz) [inline source/docs/software/basic-programming/reading-stacktraces.rst:45]: I suspect that this is not what the stack trace would look like

**Review by sciencewhiz — CHANGES_REQUESTED** [review]
In general, this needs more nuanced changes. I commented on a few places at the beginning, but these "rules" apply many more places.

- 
- Can't say Robot method or function, need to update those to say constructor
- When there's a description of a side effect that the user needs to be aware of, it's still appropriate to reference robotInit, as users can still use robotInit, we just don't want to encourage it's use with examples.
- There's cases where code samples say `void Robot()` these need to be updated since Robot isn't a method
- I suspect that the python changes aren't correct
- For the stacktraces article, I suspect the stack traces look significantly different when run from the constructor. This can't be updated just with find and replace. Per the previous point, I don't think this article is important to update as showing robotInit is ancillary to the point of explaining stack traces, but it would be bad to update the way it is that makes it wrong.

**Comment by spacey-sooty** [comment]
Resolves https://github.com/wpilibsuite/frc-docs/issues/2679

**Comment by spacey-sooty** [comment]
> Remove usage of robotInit #2679 replaced with different constructor on the FRCDocs side, should I fork allwpilib too and change the examples there too before?

allwpilib has already removed this https://github.com/wpilibsuite/allwpilib/pull/6623

**Comment by sciencewhiz** [comment]
Superseded by #2841

**Pull request closed without merging**